packages/falcon-client/src/serviceWorker/sw.js

```diff
@@ -80,22 +82,27 @@ async function getFromCacheOrNetwork(request) {
 }
 
 router.registerRoute(
-  new NavigationRoute(async ({ url }) => {
-    if (await isWaitingWithOneClient()) {
-      self.registration.waiting.postMessage({ type: 'SKIP_WAITING', payload: undefined });
-
-      return new Response('', { headers: { Refresh: '0' } }); // refresh the tab by returning a blank response
-    }
-
-    if (!CONFIG.precache) {
-      return fetch(url.href);
+  new NavigationRoute(
+    async ({ url }) => {
+      if (await isWaitingWithOneClient()) {
+        self.registration.waiting.postMessage({ type: 'SKIP_WAITING', payload: undefined });
+
+        return new Response('', { headers: { Refresh: '0' } }); // refresh the tab by returning a blank response
+      }
+
+      if (!CONFIG.precache) {
+        return fetch(url.href);
+      }
+
+      const cachedUrlKey = getCacheKeyForURL('app-shell');
+      if (!cachedUrlKey) {
+        return fetch(url.href);
+      }
+
+      return getFromCacheOrNetwork(cachedUrlKey);
+    },
+    {
+      blacklist: BLACKLIST_ROUTES.map(route => new RegExp(route))
```

Comment:
You have already entire config stored in `CONFIG` variable, so `BLACKLIST_ROUTES` is not necessary, please remove it

```suggestion
      blacklist: CONFIG.blacklistRoutes.map(route => new RegExp(route))
```

Reply:
done